internal/utils/file_write_data_utils.go

```diff
@@ -12,11 +13,19 @@ This subfolder needs to be shared across files so all functions that access it w
 
 const write_buffer_folder = "temp_write_buffers"
 
-/* Writes a file in the directory specified by temp_write_buf_dir and flushes the buffer */
+/*
+Writes a file in the directory specified by write_buffer_folder and flushes the buffer.
+This directory is meant to be cleaned up through the RemoveTempFilesDirectory() method.
+*/
 func CreateAndWriteTempFile(fileName string, data []byte) error {
-	tempWriteBufDir := filepath.Join(os.TempDir(), write_buffer_folder)
-	err := os.Mkdir(tempWriteBufDir, 0777)
-	f, err := os.Create(filepath.Join(tempWriteBufDir, fileName))
+	if _, err := os.Stat(write_buffer_folder); errors.Is(err, os.ErrNotExist) {
+		err := os.Mkdir(write_buffer_folder, 0777)
```

Comment:
Did you try looking at https://pkg.go.dev/os#MkdirAll? You could use it here to simplify this code.